nexus/auth/src/authz/context.rs

```diff
@@ -66,18 +66,22 @@ impl Authz {
 pub struct Context {
     authn: Arc<authn::Context>,
     authz: Arc<Authz>,
-    datastore: Arc<DataStore>,
+    datastore: Arc<dyn Storage>,
```

Comment:
This is another common pattern through this PR: Wherever `nexus/auth` depends on the `DataStore`, I basically use dependency injection via `dyn Storage` to not **directly** depend on the datastore. Then, the implementation of `Storage` allows `nexus/auth` to access a very limited subset of the datastore.

Turns out, that "limited interface" is only a single method. See: `nexus/auth/src/storage.rs`